pkg/executor/analyze.go

```diff
@@ -513,12 +519,17 @@ func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultsCh chan<-
 		if r := recover(); r != nil {
 			logutil.BgLogger().Error("analyze worker panicked", zap.Any("recover", r), zap.Stack("stack"))
 			metrics.PanicCounter.WithLabelValues(metrics.LabelAnalyze).Inc()
+			var job *statistics.AnalyzeJob
+			if task != nil {
+				job = task.job
+			}
 			resultsCh <- &statistics.AnalyzeResults{
 				Err: getAnalyzePanicErr(r),
-				Job: task.job,
+				Job: job,
```

Comment:
Can I ask what happens if this task is nil? Does it make sense to pass a zero job?

Reply:
In test，it can panic before getting task.

Reply:
I revert it.